Mathlib/Algebra/Lie/Extension.lean

```diff
@@ -55,16 +76,166 @@ structure Extension where
   proj : L →ₗ⁅R⁆ M
   IsExtension : IsExtension incl proj
 
-/-- The bundled `LieAlgebra.Extension` corresponding to `LieAlgebra.IsExtension` -/
-@[simps] def IsExtension.extension {i : N →ₗ⁅R⁆ L} {p : L →ₗ⁅R⁆ M} (h : IsExtension i p) :
-    Extension R N M :=
-  ⟨L, _, _, i, p, h⟩
-
 /-- A surjective Lie algebra homomorphism yields an extension. -/
 theorem isExtension_of_surjective (f : L →ₗ⁅R⁆ M) (hf : Function.Surjective f) :
     IsExtension f.ker.incl f where
   ker_eq_bot := LieIdeal.ker_incl f.ker
   range_eq_top := (LieHom.range_eq_top f).mpr hf
   exact := LieIdeal.incl_range f.ker
 
+/-- The bundled `LieAlgebra.Extension` corresponding to `LieAlgebra.IsExtension` -/
+@[simps] def Extension.ofIsExtension {i : N →ₗ⁅R⁆ L} {p : L →ₗ⁅R⁆ M}
+   (h : LieAlgebra.IsExtension i p) :
+    Extension R N M :=
+  ⟨L, _, _, i, p, h⟩
+
+end IsExtension
+
+namespace Extension
+
+variable [CommRing R] [LieRing M] [LieAlgebra R M]
```

Comment:
Just for consistency with elsewhere, could we let `L` carry the Lie algebra structure:
```suggestion
variable [CommRing R] [LieRing L] [LieAlgebra R L]
```